.github/workflows/ci.yml

```diff
@@ -0,0 +1,72 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    environment: ci
+    strategy:
+      matrix:
+        ruby-version: ['2.7', '3.0.6', '3.1', '3.2']
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```

Comment:
Is this needed? This says to fetch the entire repository, which generally isn't needed, unless there's something later on that requires that history?